guidance/d_star_lite/tests/test_d_star_lite.py

```diff
@@ -0,0 +1,82 @@
+import unittest
+import pytest
+from d_star_lite.d_star_lite import DStarLite, DSLNode
+from d_star_lite.d_star_lite import combine_nodes, compare_coordinates, distance
+import numpy as np
+
+class TestDStarLite(unittest.TestCase):
+    
+    def setUp(self):
+        # Create example DSLNodes
+        self.DSLNode1 = DSLNode(0, 0, 2.0)
+        self.DSLNode2 = DSLNode(1, 1, 3.4)
+        self.DSLNode5 = DSLNode(-1, -1, -5.8)
+
+        # Create example obstacle coordinates
+        self.ox = [1, 2, 3, 4, 5]
+        self.oy = [0, 0, 0, 0, 0]
+
+        # Create example dsl object
+        self.dsl = DStarLite(self.ox, self.oy)
+
+    def tearDown(self):
+        pass
+
+    def test_combine_DSLNodes(self):
+        # Test the combine_DSLNodes function
+        result = combine_nodes(self.DSLNode1, self.DSLNode2)
+        self.assertEqual(result.x, 1)
+        self.assertEqual(result.y, 1)
+        self.assertEqual(result.cost, 5.4)
+
+        result = combine_nodes(self.DSLNode2, self.DSLNode5)
+        self.assertEqual(result.x, 0)
+        self.assertEqual(result.y, 0)
+        self.assertEqual(result.cost, -2.4)
+
+    def test_compare_coordinates(self):
+        # Test the compare_coordinates function
+        result = compare_coordinates(self.DSLNode1, self.DSLNode2)
+        self.assertEqual(result, False)
+
+        result = compare_coordinates(self.DSLNode1, self.DSLNode1)
+        self.assertEqual(result, True)
```

Comment:
Tests 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 

Don't be afraid of having *MANY* defs in a test file; having more, very specifically named tests instead of combining them into one is usually easier to read through should any of them fail. 

For example, this could be split into:

```python
def test_compare_coordinates_is_false_for_two_different_nodes(self):
    ...
``` 
and
```python
def test_compare_coordinates_is_true_for_two_equal_nodes(self):
    ...
``` 

same applies to every other method here that combines multiple tests